src/time_integration/methods_SSP.jl

```diff
@@ -280,12 +280,14 @@ function modify_dt_for_tstops!(integrator::SimpleIntegratorSSP)
             integrator.dt = integrator.tdir *
                             min(abs(integrator.dt), abs(tdir_tstop - tdir_t)) # step! to the end
         elseif iszero(integrator.dtcache) && integrator.dtchangeable
-            integrator.dt = integrator.tdir * abs(tdir_tstop - tdir_t)
+            integrator.dt = integrator.tdir *
+                            min(abs(integrator.dt), abs(tdir_tstop - tdir_t)) # step! to the end
         elseif integrator.dtchangeable && !integrator.force_stepfail
             # always try to step! with dtcache, but lower if a tstop
             # however, if force_stepfail then don't set to dtcache, and no tstop worry
             integrator.dt = integrator.tdir *
-                            min(abs(integrator.dtcache), abs(tdir_tstop - tdir_t)) # step! to the end
+                            min(abs(integrator.dt), abs(integrator.dtcache),
```

Comment:
I'm not really sure about that third case here. In the simulation I tested, this did not occur. To be safe, I added `abs(integrator.dt)`. What do you say about that @amrueda

Reply:
Has been settled after [49f3f15](https://github.com/trixi-framework/Trixi.jl/pull/1919/commits/49f3f15f8034e3933a68e3e57f64205f20085cc8)